From 6de2d56483d6141f26c28ebd3f5e9085ecf24b0a Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Tue, 28 Jul 2026 11:50:21 +0800
Subject: [PATCH 334/334] AOSCOS: drm/ttm: Replace dev->dma_coherent with
 dev_dma_coherent()

In C, bitfields are not necessarily safe to modify from multiple
threads without locking. Switch "dma_coherent" over to the "flags"
field so modifications are safe.

Link: https://git.kernel.org/torvalds/c/3e2c1e213ac2
Fixes: 74b787c07601 ("BACKPORT: FROMLIST: drm/ttm: downgrade cached to write_combined when snooping not available")
---
 drivers/gpu/drm/ttm/ttm_device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/ttm/ttm_device.c b/drivers/gpu/drm/ttm/ttm_device.c
index 7fad8dcd1ecfe..376199665a8b5 100644
--- a/drivers/gpu/drm/ttm/ttm_device.c
+++ b/drivers/gpu/drm/ttm/ttm_device.c
@@ -247,7 +247,7 @@ int ttm_device_init(struct ttm_device *bdev, const struct ttm_device_funcs *func
 #if defined(CONFIG_ARCH_HAS_SYNC_DMA_FOR_DEVICE) || \
 	defined(CONFIG_ARCH_HAS_SYNC_DMA_FOR_CPU) || \
 	defined(CONFIG_ARCH_HAS_SYNC_DMA_FOR_CPU_ALL)
-	bdev->dma_coherent = dev->dma_coherent;
+	bdev->dma_coherent = dev_dma_coherent(dev);
 #endif
 
 	return 0;
-- 
2.55.0

